src/globus_sdk/base.py

```diff
@@ -361,42 +262,23 @@ def _request(
         :type params: dict
         :param headers: HTTP headers to add to the request
         :type headers: dict
-        :param json_body: Data which will be JSON encoded as the body of the request
-        :type json_body: dict
-        :param text_body: Either a raw string that will serve as the request body, or a
-            dict which will be HTTP Form encoded
-        :type text_body: str or dict
+        :param data: Data to send as the request body. May pass through encoding.
+        :type data: dict or string
+        :param encoding: A way to encode request data. "json", "form", and "text"
+            are all valid values. Custom encodings can be used only if they are
+            registered with the transport. By default, strings get "text" behavior and
+            all other objects get "json".
+        :type encoding: string
         :param response_class: Class for response object, overrides the client's
             ``default_response_class``
         :type response_class: class
-        :param retry_401: Retry on 401 responses with fresh Authorization if
-            ``self.authorizer`` supports it
-        :type retry_401: bool
 
         :return: :class:`GlobusHTTPResponse \
         <globus_sdk.response.GlobusHTTPResponse>` object
         """
-        # copy
-        rheaders = dict(self._headers)
-        # expand
-        if headers is not None:
-            rheaders.update(headers)
-
-        if json_body is not None:
-            assert text_body is None
-            text_body = json.dumps(json_body)
-            # set appropriate content-type header
-            rheaders.update({"Content-Type": "application/json"})
-
-        # add Authorization header, or (if it's a NullAuthorizer) possibly
-        # explicitly remove the Authorization header
-        if self.authorizer is not None:
-            log.debug(
-                "request will have authorization of type {}".format(
-                    type(self.authorizer)
-                )
-            )
-            self.authorizer.set_authorization_header(rheaders)
+        # prepare data...
+        # copy headers if present
+        rheaders = {**headers} if headers else {}
```

Comment:
Interesting, I hadn't seen the unpacking syntax used for copying before. Was there a particular reason you adopted it here?

Reply:
The `**` syntax is really handy when you're copying and modifying a dict to add a single attribute. e.g.
`d = {"x": 1, **d}`

Because it gets used like that, I've started to prefer it as a syntax for plain copying. That way, the two usages are consistent throughout a codebase.
Plus, I think it's more obvious that this is a shallow copy than `dict.copy` or `copy.copy`. The syntax agrees with splat-expansion in other contexts (e.g. keyword args) in doing a shallow copy.

This is the sort of stylistic thing that I don't think is a big deal -- certainly not enough to try to add a lint for it. I personally prefer this way of writing it, and I have my reason for that, but it doesn't run any deeper than that.